pythainlp/corpus/common.py

```diff
@@ -75,6 +78,31 @@ def provinces() -> frozenset:
 
     return _THAI_THAILAND_PROVINCES
 
+def provinces_all() -> list:
```

Comment:
maybe we can combine this function with `provinces()`?
To me, because these two functions are quite similar, we can have an argument letting the user to choose which list  he/she wants.

For example, the signatures could be: `province()` and `province(details=True)`.

Reply:
OK.

Reply:
It's tricky here.

`provinces()` was designed to populate a set of province names in the first run (when `_THAI_THAILAND_PROVINCES` is still empty, it will read the names from a file with a name `_THAI_THAILAND_PROVINCES_FILENAME`) and for any other subsequent call it will just return `_THAI_THAILAND_PROVINCES` - to save processing time and avoid IO operation).

The modification breaks this.

May be we can modify the function in this way:
- Has only one file (the CSV one) of Thai names and romanized names (province Ministry of Interior code, province abbreviation, province postal code, etc.)
- In the first call, populate as set (or sorted list?) of name pairs to `_THAI_THAILAND_PROVINCES`.
- In any other subsequent call, just use the existing `_THAI_THAILAND_PROVINCES`.
- The return type/format will be determined by value in `details=`

Reply:
@bact OK. I updated code.